### PR TITLE
feat: Support rebase only when newer version has been found

### PR DESCRIPTION
**Pull request by bobvandevijver**
## Changes:

This change should implement my feature request in https://github.com/renovatebot/renovate/discussions/10564. 

## Context:

Sometimes, you only want renovate to rebase/retry a PR whenever the external dependency has been changed in order to save valuable computing/pipeline resources. When using the new `rebaseWhen=newer-version` renovate will only update existing branches when it has found a newer version of the dependency.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -1791,7 +1792,7 @@ Possible values and meanings:
 - It could result in a broken base branch if two updates are merged one after another without testing the new versions together
 - If you have enforced that PRs must be up-to-date before merging (e.g. using branch protection on GitHub), then automerge won't be possible as soon as a PR gets out-of-date but remains non-conflicted
 
-It is also recommended to avoid `rebaseWhen=never` as it can result in conflicted branches with outdated PR descriptions and/or status checks.
+It is also recommended avoiding `rebaseWhen=never` as it can result in conflicted branches with outdated PR descriptions and/or status checks.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:1795]:
```suggestion
We recommend you avoid setting `rebaseWhen` to `never` as you can end up with conflicted branches with outdated PR descriptions and/or status checks.
```

File: docs/usage/updating-rebasing.md [hunk 2]
```diff
@@ -50,6 +50,11 @@ This way:
 - Each Renovate branch will always have 1 and only 1 commit
 - The newest version will be based off the latest base branch commit at the time
 
+When using the default configuration (`"rebaseWhen": "auto"`), this will happen automatically.
```
Comment (HonkingGoose) [inline docs/usage/updating-rebasing.md:53]:
```suggestion
When you use the default configuration (`"rebaseWhen": "auto"`), this happens automatically.
```

File: docs/usage/updating-rebasing.md [hunk 3]
```diff
@@ -50,6 +50,11 @@ This way:
...
+However, if you only want Renovate to rebase when a newer version of the dependency "foo" is available, you can configure `"rebaseWhen": "newer-version"`.
```
Comment (HonkingGoose) [inline docs/usage/updating-rebasing.md:54]:
```suggestion
However, if you want Renovate to rebase only when a newer version of the dependency "foo" is available, you can configure `"rebaseWhen": "newer-version"`.
```

File: docs/usage/updating-rebasing.md [hunk 4]
```diff
@@ -50,6 +50,11 @@ This way:
...
+
+This behaviour is fully disabled by configuring `"rebaseWhen": "never"` (not recommended).
```
Comment (HonkingGoose) [inline docs/usage/updating-rebasing.md:56]:
```suggestion
If you don't want Renovate to rebase branches, you can configure `"rebaseWhen": "never"` (not recommended).
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
I think this is easier to read and understand. 😉

**Review by viceice — COMMENTED** [review]
I think this needs a real repo run even it's not easy to do.

**Comment by bobvandevijver** [comment]
> I think this needs a real repo run even it's not easy to do.

I agree, I'm just having a hard time finding the required to create it. I'm working on it!

**Comment by rarkins** [comment]
@viceice can you think of any simpler ways we could achieve this? I'd love to avoid complicating this interface if possible

**Comment by bobvandevijver** [comment]
@rarkins / @viceice I just tested this on a real repository, and it doesn't seem to work as I had hoped. Probably caused because I simply do not have enough insight in the functioning inside the update logic. Maybe one of you could take a look?

**Pull request closed without merging**